src/app.js

```diff
@@ -29,43 +46,107 @@ passport.deserializeUser(function(obj, cb) {
   cb(null, obj);
 });
 
-const app = express();
-app.use(express.static('public'))
+app.use(express.static('public'));
 app.use(require('morgan')('combined'));
 app.use(require('body-parser').urlencoded({ extended: true }));
-app.use(require('express-session')({ secret: 'keyboard cat', resave: true, saveUninitialized: true }));
+app.use(require('express-session')({ secret: 'keyboard cat', resave: false, saveUninitialized: false }));
 
 app.use(passport.initialize());
-app.use(passport.session());
+// app.use(passport.session());
 
-app.get('/', (request, response) => {
-  response.send('index.html');
+app.get('/login/twitter', passport.authenticate('twitter'), (request, response) => {
+  response.send({ hello: 'this is the auth route' });
 });
 
-app.get('/login', (request, response) => {
-  response.sendStatus(200);
-});
+app.get(
+  '/oauth/callback',
+  passport.authenticate('twitter', { failureRedirect: '/login/twitter' }),
+  (request, response) => {
+    const sessionId = request.sessionID;
+    const sessionData = request.sessionStore.sessions[sessionId];
+    const oAuthData = JSON.parse(sessionData)['oauth:twitter'];
 
-app.get('/login/twitter', passport.authenticate('twitter'));
+    const userData = request.user._json;
 
-app.get('/oauth/callback',
-  passport.authenticate('twitter', { failureRedirect: '/login' }),
-  (request, response) => {
-    response.redirect('exp://8q-3ig.kozlowskicd.react-native-spike.exp.direct:80');
+    const savedUserData = {
+      userId: userData.id,
+      userName: userData.name,
+      userScreenName: userData.screen_name,
+      photoLink: request.user.photos[0].value,
+      oAuthToken: request.query.oauth_token,
+      oAuthVerifier: request.query.oauth_verifier,
+      oAuthTokenSecret: oAuthData.oauth_token_secret,
+    };
+
+    // add to database
```

Comment:
:) you know what I'm going to write right? no commented out messages!

Reply:
removed!